cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/batch/BatchSparkPipelineDriver.java

```diff
@@ -84,10 +90,12 @@ public void run(DatasetContext context) throws Exception {
                                              BatchPhaseSpec.class);
 
     try (InputStream is = new FileInputStream(sec.getLocalizationContext().getLocalFile("HydratorSpark.config"))) {
-      sourceFactory = SparkBatchSourceFactory.deserialize(is);
-      sinkFactory = SparkBatchSinkFactory.deserialize(is);
-      DataInputStream dataInputStream = new DataInputStream(is);
-      stagePartitions = GSON.fromJson(dataInputStream.readUTF(), MAP_TYPE);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
```

Comment:
This hasn't been fixed yet? I had a previous comment using `Files.newBufferedReader` in the `try` statement directly.